File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/dbhelpers/DBProvider.java
package app.zxtune.fs.dbhelpers;

import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DBProvider {

  private static final String TAG = DBProvider.class.getName();

  private final SQLiteOpenHelper delegate;

  public DBProvider(SQLiteOpenHelper delegate) {
    this.delegate = delegate;
  }

  public final SQLiteDatabase getWritableDatabase() {
    return delegate.getWritableDatabase();
  }

  public final SQLiteDatabase getReadableDatabase() {
    return delegate.getReadableDatabase();
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/dbhelpers/DBProvider.kt
package app.zxtune.fs.dbhelpers

import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DBProvider(private val delegate: SQLiteOpenHelper) {

    init {
        DBStatistics.send(delegate)
    }

    fun close() = delegate.close()

    val writableDatabase: SQLiteDatabase
        get() = delegate.writableDatabase

    val readableDatabase: SQLiteDatabase
        get() = delegate.readableDatabase
}
